File: CDCAT_ONLINE/Eta.m
function etas = Eta(alpha,Q)
%alpha is a person by item matrix
%Q is the Q matrix
%etas is a person by item matrix

eta0 = alpha*Q';
sumQ = sum(Q,2)';
sumQ_ = Q*Q';
etas = eta0==sumQ(ones(size(eta0,1),1),:);

File: Sequetial Search Q validation/functions/Eta.R
###############################################################################
# Eta:                                                                        #
#                                                                             #
# Compute eta values for each examinee, each item                             #
#                                                                             #
# Input:                                                                      #
# (1) alpha  : alpha for N examinee, N by k                                   #
# (2) Q      : Q matrix                                                       #
#                                                                             #
# Output:                                                                     #
# (1) eta    : whether this examinee has mastered all the required attributes # 
###############################################################################



Eta <- function(alpha,Q) {
  
  eta0 <- alpha %*% t(Q)
  sumQ <- rowSums(Q)
  eta  <- t(apply(eta0,1, function(x) (x==sumQ)*1))
 
  return(eta)
}
